src/web-blocks/pagination.tsx

```diff
@@ -0,0 +1,177 @@
+
+    // Previous button
+    buttons.push(
+      <a
+        key="prev"
+        href={currentPageNum > 1 ? getPageUrl(strategy, currentPageNum - 1) : "#"}
+        {...arrowButtonStyles}
+        className={cn(
+          get(arrowButtonStyles, "className", ""),
+          currentPageNum <= 1 && !inBuilder && "pointer-events-none cursor-not-allowed opacity-70",
+        )}>
+        <ChevronLeftIcon />
+      </a>,
+    );
+
+    // Page buttons
+    for (let i = startPage; i <= endPage; i++) {
+      const isActive = i === currentPageNum || (inBuilder && i === 1);
+      buttons.push(
+        <a
+          key={i}
+          href={getPageUrl(strategy, i)}
+          {...(isActive ? activeItemStyle : inactiveItemStyle)}
+          className={cn(
+            get(isActive ? activeItemStyle : inactiveItemStyle, "className", ""),
+            isActive && "pointer-events-none",
+          )}>
+          {i}
+        </a>,
+      );
+    }
+
+    // Next button
+    buttons.push(
+      <a
+        key="next"
+        href={currentPageNum < totalPages ? getPageUrl(strategy, currentPageNum + 1) : "#"}
+        {...arrowButtonStyles}
+        className={cn(
+          get(arrowButtonStyles, "className", ""),
+          currentPageNum >= totalPages && !inBuilder && "pointer-events-none cursor-not-allowed opacity-70",
+        )}>
+        <ChevronRightIcon />
+      </a>,
+    );
+
+    return buttons;
+  };
+
+  return (
+    <div {...blockProps} {...styles}>
+      {renderPaginationButtons()}
+    </div>
+  );
+};
+
+export const PaginationBlockConfig = {
+  type: "Pagination",
+  icon: ButtonIcon,
+  label: "Pagination",
+  category: "core",
+  group: "basic",
+  ...registerChaiBlockSchema({
+    properties: {
+      styles: StylesProp("flex items-center justify-center gap-2"),
+      activeItemStyle: StylesProp(
+        "flex h-8 w-8 font-medium items-center justify-center select-none bg-blue-500 text-white rounded-full text-xs",
+      ),
+      inactiveItemStyle: StylesProp(
+        "flex h-8 w-8 font-medium items-center justify-center select-none bg-gray-200 text-gray-700 rounded-full text-xs hover:bg-gray-300",
+      ),
+      arrowButtonStyles: StylesProp(
+        "flex h-8 w-8 font-medium items-center justify-center select-none bg-gray-200 text-gray-700 rounded-full text-xs hover:bg-gray-300",
+      ),
+      strategy: {
+        type: "string",
+        title: "Pagination Strategy",
+        default: "query",
+        oneOf: [
+          { enum: ["query"], title: "Query Parameters" },
+          { enum: ["segment"], title: "URL Segment" },
+        ],
+      },
+      visibleButtonCounts: {
+        type: "number",
+        title: "Visible Button Count",
+        default: 3,
+        minimum: 1,
+        maximum: 7,
+      },
+      totalItems: closestBlockProp("Repeater", "totalItems"),
+      limit: closestBlockProp("Repeater", "limit"),
+      currentPage: closestBlockProp("Repeater", "currentPage"),
```

Comment:
current page is not available on Repeater block. we should read the query params and detect current page